### PR TITLE
Integrating Fluent2 TableViewCell into main.

### PR DESCRIPTION
**Pull request by rdeassis**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

This change merges the Fluent2 version of the TableViewCell into main with support for design tokens.
For reference, the changes in the fluent2-tokens branch associated with the TableViewCell are listed below:

- #1097 
- #1088
- #1082
- #1067
- #1062
- #1056 
- #1044
- #1002 
- #998 
- #978 

### Verification

- Clean build of dependent partner apps (devmain)
- Test scenarios exercised through the Demo app.

|                      **Before**                 |                   **After**                   |
|----------------------------------------------|--------------------------------------------|
| image | image |
| image | image |
| image | image |
| image | image |
| N/A | image |
| image | image |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [x] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [x] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [x] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [x] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1059)

### CODE REVIEW

File: ios/FluentUI/Other Cells/ActionsCell.swift [hunk 1]
```diff
@@ -95,9 +129,9 @@ open class ActionsCell: UITableViewCell {
     private var action1Type: ActionType = .regular
     private var action2Type: ActionType = .regular
 
-    private let topSeparator = Separator(style: .default, orientation: .horizontal)
-    private let bottomSeparator = Separator(style: .default, orientation: .horizontal)
-    private let verticalSeparator = Separator(style: .default, orientation: .vertical)
+    private let topSeparator = MSFDivider()
```
Comment (huwilkes) [inline ios/FluentUI/Other Cells/ActionsCell.swift:132]: There seems to be a cocoapods issue with not being able to find MSFDivider

Comment (rdeassis) [reply ios/FluentUI/Other Cells/ActionsCell.swift:132]: Yes. Just noticed it. Validating locally and will push the fix for it soon.
There is no subspec for the divider and it should be a dependency for the TableViewCell.

File: MicrosoftFluentUI.podspec [hunk 2]
```diff
@@ -287,6 +293,7 @@ fi', :execution_position => :before_compile }
 
   s.subspec 'TableView_ios' do |tableview_ios|
     tableview_ios.platform = :ios
+    tableview_ios.dependency 'MicrosoftFluentUI/Divider_ios'
     tableview_ios.dependency 'MicrosoftFluentUI/Label_ios'
     tableview_ios.dependency 'MicrosoftFluentUI/Separator_ios'
```
Comment (huwilkes) [inline MicrosoftFluentUI.podspec:298]: Would it be possible to clean up the Separator at the same time, or should that happen in another PR?

File: ios/FluentUI/Table View/TableViewTokens.swift [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    open var labelAccessoryViewMarginLeading: CGFloat { globalTokens.spacing[.xSmall] }
+
+    /// The trailing margin for the labelAccessoryView.
+    open var labelAccessoryViewMarginTrailing: CGFloat { globalTokens.spacing[.xSmall] }
+
+    /// The leading margin for the customAccessoryView.
+    open var customAccessoryViewMarginLeading: CGFloat { globalTokens.spacing[.xSmall] }
+
+    /// The minimum vertical margin for the customAccessoryView.
+    open var customAccessoryViewMinVerticalMargin: CGFloat = 6
```
Comment (huwilkes) [inline ios/FluentUI/Table View/TableViewTokens.swift:94]: This also needs to have #1044 applied

Comment (rdeassis) [reply ios/FluentUI/Table View/TableViewTokens.swift]: Thanks for catching it. Just cherry-picked that change.

File: MicrosoftFluentUI.podspec (hunk 2 above)
Comment (rdeassis) [reply MicrosoftFluentUI.podspec:298]: I think we can tackle it in a separate work item. This one is already too big/complex.

**Review by sophialee0416 — APPROVED** [review]

**Comment by sophialee0416** [comment]
Thanks in advance for adding in #1062 :)

**Review by mischreiber — APPROVED** [review]

**Comment by harrieshin** [comment]
can we actually do the before and after page of the tableviewcell in ipad splitmode elevated colors which seems to be where majority of the bug is? the current screenshot is bit hard to tell the before and after per light/dark theme

File: MicrosoftFluentUI.podspec (hunk 2 above)
Comment (sophialee0416) [reply MicrosoftFluentUI.podspec]: Resolved offline; we're removing divider in this PR and leaving separator in for the time being

**Review by harrieshin — APPROVED** [review]

**Pull request merged**